Mathlib/Algebra/CharP/Basic.lean

```diff
@@ -126,6 +126,11 @@ theorem add_pow_char_pow [CommSemiring R] {p : ℕ} [Fact p.Prime] [CharP R p] {
   add_pow_char_pow_of_commute _ _ _ (Commute.all _ _)
 #align add_pow_char_pow add_pow_char_pow
 
+theorem add_pow_eq_add_pow_mod_mul_pow_add_pow_div
+    [CommSemiring R] {p : ℕ} [Fact p.Prime] [CharP R p] {n : ℕ} (x y : R) :
+    (x + y) ^ n = (x + y) ^ (n % p) * (x ^ p + y ^ p) ^ (n / p) := by
+  rw [← add_pow_char, ← pow_mul, ← pow_add, Nat.mod_add_div]
```

Comment:
Oops I missed this, I will do it yes

Reply:
@YaelDillies I think `p` doesn't have to be explicit since `x` and `y` would infer `R` which should uniquely infer `p` - at least mathematically the second part is true?

Reply:
Mathematically, yes. In practice, not really 😢